src/Cake.Core/IO/LaunchInfo.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Cake.Core.IO
+{
+    /// <inheritdoc/>
+    public sealed class LaunchInfo : ILaunchInfo
```

Comment:
Lets make this class `internal` as build scripts shouldn't need to use this directly.

Reply:
@augustoproiete Would bridge / frosting need this, in that case public

Reply:
@devlead Hmm... Good point. I thought the `ILaunchInfo` interface was enough and they could have their own implementations, but we're sharing a lot of concrete implementations already.

[Frosting uses it](https://github.com/cake-build/cake/pull/3132/files#diff-a4fe3eef0c77283c6481738eb652efa3f56018d9a316dbb4575b0474038abd9fR85), so it's safe to assume Bridge would too.

If everything will be public, maybe better to rename to `ICakeLaunchInfo`/`CakeLaunchInfo` for consistency? (CakeEnvironment, CakePlatform, CakeRuntime, ...)